test/integration/apiserver/certreload/aggregator_certreload_test.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package certreload
+
+import (
+	"context"
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	genericapiserveroptions "k8s.io/apiserver/pkg/server/options"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/cert"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	aggregatorclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	kastesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	apiserverintegration "k8s.io/kubernetes/test/integration/apiserver"
+	"k8s.io/kubernetes/test/integration/framework"
+	testutil "k8s.io/kubernetes/test/utils"
+	sampleserver "k8s.io/sample-apiserver/pkg/cmd/server"
+)
+
+func TestAggregatorCertReload(t *testing.T) {
+	proxyCA, proxyClientKey, proxyClientCert, err := newCAWithClientCert()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var proxyClientKeyPath string
+	var proxyClientCertPath string
+	var servingCertPath string
```

Comment:
I'm initializing proxyClientKeyPath and proxyClientCertPath at the beginning of the test.

I'm initializing servingCertPath inside ModifyServerRunOptions because I don't have access to it. It is populated in kastesting.StartTestServerOrDie function